### ISSUE TITLE
How to fine tune the detector and the recogniser for a custom dataset?

### ISSUE THREAD

**Issue by Pusse-01** [faustomorales/keras-ocr#194]
Can anyone explain what is the format of the data or how is the dataset should be feeded to the model to fine tune. I have a set of images which have been labeled each and every character by drawing bounding boxes. So how I could fine tune the model using this data set to increase the accuracy of my model.

**Comment by JackSadigov** [comment]
Hi. 
I am also looking for the same here. 
 I tried Convert PASCAL dataset to TFRecord for object detection in TensorFlow explained [here](https://androidkt.com/convert-pascal-dataset-to-tfrecord-for-object-detection-in-tensorflow/ ) but got an error with ''annotations has no object".

**Comment by seansteel3** [comment]
@Pusse-01 and @JackSadigov here is an excellent guide that will take you through retraining: https://keras-ocr.readthedocs.io/en/latest/examples/end_to_end_training.html

There are two open issues related to:
Detector Batch size: https://github.com/faustomorales/keras-ocr/issues/218
Recognizer retraining: https://github.com/faustomorales/keras-ocr/issues/231

For the recognizer retraining Oadon's solution worked for me, I did not attempt geo-tp's but it probably works too. Obviously make sure you're data is in the same format as the guide (or edit the guide accordingly) to retrain.
